src/Mod/Part/App/AttachExtension.cpp

```diff
@@ -477,6 +482,164 @@ bool AttachExtension::extensionHandleChangedPropertyName(Base::XMLReader& reader
     return App::DocumentObjectExtension::extensionHandleChangedPropertyName(reader, TypeName, PropName);
 }
 
+void AttachExtension::handleLegacyTangentPlaneOrientation()
+{
+    // check attachment mode
+    if (_props.attacher->mapMode != mmTangentPlane) {
+        return;
+    }
+
+    // check stored document program version (applies to v1.0 and earlier only)
+    int major, minor;
+    if (sscanf(getExtendedObject()->getDocument()->getProgramVersion(), "%d.%d", &major, &minor) != 2) {
+        return;
+    }
+    if (major > 1 || (major == 1 && minor > 0)) {
+        return;
+    }
+
+    // check for an App::Plane support object exists
+    App::GeoFeature* geof = nullptr;
+    for (auto obj : this->AttachmentSupport.linkedObjects()) {
+        if (obj->isDerivedFrom(Base::Type::fromName("App::Plane"))) {
+            geof = dynamic_cast<App::GeoFeature*>(obj);
+            break;
+        }
+    }
+    if (!geof) {
+        return;
+    }
+
+    /*
+     * determine which global axis (X/Y/Z) is closest to the plane normal
+     */
+
+    // extract the plane normal
+    Base::Vector3d norm;
+    geof->Placement.getValue().getRotation().multVec(Base::Vector3d(0.0, 0.0, 1.0), norm);
+    gp_Dir normal(norm.x, norm.y, norm.z);
+
+    // determine the dominant global axis
+    std::array<double, 3> cosXYZ = {
+        fabs(normal.Dot(gp_Dir(1, 0, 0))),  // normal dot global X axis
+        fabs(normal.Dot(gp_Dir(0, 1, 0))),  // normal dot global Y axis
+        fabs(normal.Dot(gp_Dir(0, 0, 1)))   // normal dot global Z axis
+    };
+    int axis = std::max_element(cosXYZ.begin(), cosXYZ.end()) - cosXYZ.begin();
```

Comment:
Using ranges and preventing a conversion from long to int

```suggestion
    std::size axis = std::ranges::max_element(cosXYZ) - cosXYZ.begin();
```

Reply:
Thanks — I updated it to use std::distance to compute the index:
```
std::size_t axis = std::distance(cosXYZ.begin(), std::ranges::max_element(cosXYZ));
```